projects/saturn/src/main/java/io/fairspace/saturn/services/metadata/MetadataService.java

```diff
@@ -72,7 +72,9 @@ public Model get(String subject, boolean withValueProperties) {
                     .forEach(property -> m.listStatements(null, property, resource)
                             .filterKeep(stmt -> permissions.canReadMetadata(stmt.getSubject()))
                             .filterDrop(stmt -> stmt.getSubject().hasProperty(FS.dateDeleted))
-                            .forEachRemaining(stmt -> {
+                            .toList()
+                            .subList(0, 21)
```

Comment:
this doesn't seem like a performance fix - it will still need to fetch them all, transform to list and only then take a sublist...
Isn't there some Jena way to limit number of results on iterator level? I see there is a `limit()` method on stream<Resource>, not sure how to do for stream<Property> when filtering

Reply:
Well it is because the data is not send to the client, which takes most time. I can make it even faster indeed by not doing the 'toList'.

Reply:
would be nice if possible. But is there some Jena way that would allow replacing toList? I wanted to suggest something, but couldn't find it easily

Reply:
Also it looked like this change caused tests to fail...